### ISSUE TITLE
Cannot interact with 2 open JNLPs (out of which the first one triggers browser that triggers second one)

### ISSUE THREAD

**Issue by iamalittletester** [appium/appium#11135]
## The problem
I am currently creating an end to end automated scenario, that requires to: launch a Java desktop application through JNLP; from this application a browser will be launched which in turn will launch another JNLP process which is another application built with OracleForms; after this is done successfully, the two JNLPs run simultaneously on the machine; from the lastly opened JNLP, by clicking a button, a browser is launched and the testing needs to continue inside the browser.
There are two problems regarding this flow:
1. Currently it is impossible in my test to interact with the second JNLP instance. Appium only sees one instance, and there is only 1 window handle identified by Appium. I have tried the workaround described in https://github.com/Microsoft/WinAppDriver/blob/v1.0-RC2/README.md#attaching-to-an-existing-app-window but still no success (i tried to attach the process of the second JNLP to the first one).
2. Assuming a fix exists for point 1, i am not sure i can then interact with the newly opened browser from Appium, as it is launched by the JNLP application and not by opening a URL directly from the test. The end goal is to be able to test inside this newly open browser by going through the flow mentioned above.
Please read the "Details" section for further details.  

## Environment
Appium version: java-client version 6.1.0
Language used: Java 
Desktop/OS version: Windows 10
Node.js version: 8.11.3
Platform under test: Windows 10 PC
Device under test: Windows application

## Details
The steps that need to be automated are as follows:
1. Launch a Java windows application through JNLP. This is done successfully and the application can be interacted with from Appium.
2. By clicking on a button on this application, a new application is triggered to open by first opening a browser window which triggers the second application to open through JNLP.  So basically, the flow is: clicking on first application button opens > browser, which opens > second application. This second application is built using OracleForms. Both desktop applications remain open at this time. The browser needs to remain open for the second application to also remain open. This browser window does not need interacting with.

The problem:
3. At this step, Appium only sees one window handle for the JNLP applications. I can only interact with the first open JNLP. I cannot interact with the second JNLP. Both appear as processes running under the Java launcher process in Task Manager.
What is needed: to be able to control the second JNLP also, in order to launch the new browser, inside which interactions with the pages are needed.

If this problem would be fixed, the next problem to solve is:
4. From the second JNLP, by clicking on a button, a new browser instance will open that needs to be interacted with.

Unfortunately i cannot provide any logs or code for this issue. 
Any help would be appreciated. Thank you.

**Comment by jlipps** [comment]
Hmm, our team doesn't know much about WinAppDriver. Maybe @yodurr or someone on his team has an idea about this?

**Comment by yodurr** [comment]
If you create the DesktopSession 
https://github.com/Microsoft/WinAppDriver/blob/v1.0-RC2/README.md#creating-a-desktop-session
you should be able to search for any of your app Windows and attach to them for testing using https://github.com/Microsoft/WinAppDriver/blob/v1.0-RC2/README.md#attaching-to-an-existing-app-window.

You might try pointing the test at WinAppDriver directly by running winappdriver.exe directly on the command line and pointing your tests at WinAppDriver.  The behavior shouldn't be different from using Appium but if it is that would be a good bug for us to investigate.

I'm not familiar with JNLP - if you know of any samples online that are similar to what your app is doing I can take a look.

**Comment by iamalittletester** [comment]
Thanks @yodurr for the tips. I finally figured out how to get it working. So what i did differently from my initial implementation:
- i started the WinAppDriver instance as you suggested from the exe file
- initially when creating the session, in the 'new WindowsDriver' i had not passed in the URL of the running WinAppDriver instance, since i did not start it. I used the 'new' command only with the capabilities. Now i changed this to use the newly spawned WinAppDriver instance
- i initially used the "Classic Windows Application" . I now switched to the "Desktop Session" approach. And this way my flow works, and i don't even need to attach any windows to an existing process. I was not aware of this last one.
Thank you so much for the help. Much appreciated!
